### PR TITLE
New specs tests

### PR DESCRIPTION
**Pull request by Reallife**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | No

### Added

- Tests for specifications using `nodejs` and `jest` [#137]

### Changed

- `Makefile` goals `test` and `install` [#137]

[#137]:https://github.com/avtocod/specs/issues/137

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    branches:
+      - master
+    tags-ignore:
+      - '**'
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0' # once in a week, docs: <https://git.io/JvxXE#onschedule>
+
+env:
+  NODE_VERSION: '14'
```
Comment (tarampampam) [inline .github/workflows/tests.yml:14]:
```suggestion
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v1 # Action page: <https://github.com/actions/setup-node>
+        with:
+          node-version: ${{ env.NODE_VERSION }}
```
Comment (tarampampam) [inline .github/workflows/tests.yml:27]:
```suggestion
          node-version: 14
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,13 +9,18 @@ The format is based on [Keep a Changelog][keepachangelog] and this project adher
 ### Added
 
 - Composer dependency `avtocod/specs-php`  [#137]
+- Tests for specifications using `nodejs` and `jest` [#137]
+
+### Changed
+
+- `Makefile` goals `test` and `install` [#137]
+- Start using `node:14-alpine` docker image instead of `php:7.2.5-alpine` [#137]
```
Comment (tarampampam) [inline CHANGELOG.md:17]:
```suggestion
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "avtocod-specs",
+  "version": "3.46.0",
```
Comment (tarampampam) [inline package.json:3]:
```suggestion
  "version": "0.0.0",
```

File: tests/vehicles/vehicles_models.test.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import * as path from 'path';
+import { groups_list, specs_root_dir, required_files_map } from "../helpers";
+import { VehicleMark, VehicleModel } from "./vehicles";
+
+const models_file_names = required_files_map.vehicles.filter(filename => {
+    return ['marks.json', 'types.json'].indexOf(filename) === -1;
+});
+const expected_items_count: {[key: string]: {[k: string]: number}} = {
```
Comment (tarampampam) [inline tests/vehicles/vehicles_models.test.ts:8]: спорно, так как добавление нового (корректного формата) будет ломать запуск тестов

Comment (Reallife) [reply tests/vehicles/vehicles_models.test.ts]: Removed

File: fields/default/fields_list.json [hunk 6]
```diff
@@ -3174,7 +3174,7 @@
         "path": "ads.history.items[].photos.grz",
         "description": "История объявлений: Список ГРЗ c URI фотографии, на которых они распознаны",
         "types": [
-            "array"
+            "object"
```
Comment (Reallife) [inline fields/default/fields_list.json:3177]: @tarampampam Found an error. This field wass added in `v3.2.0` in [this PR](https://github.com/avtocod/specs/pull/48/files). Actually this field contains object, not array

File: tests/fields/fields_list.test.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                        case 'object':
+                            expect(report_field_value).toBeObject();
+                            break;
+                        case 'array':
+                            expect(report_field_value).toBeArray();
+                            break;
+                        case 'boolean':
+                            expect(report_field_value).toBeBoolean();
+                            break;
+                        case 'float':
```
Comment (Reallife) [inline tests/fields/fields_list.test.ts:91]: @tarampampam 
In this case we can't detect real data type (`float` or `integer`) for values such as `100.0`, e.g 
```js
Number.isInteger(100.0) // true
parseFloat(100.0) // 100
```
 Also json-schema has no data type `float`. What will be better solution for this case?

File: fields/default/fields_list.json (hunk 6 above)
Comment (tarampampam) [reply fields/default/fields_list.json:3177]: Nice job!

File: tests/fields/fields_list.test.ts (hunk 7 above)
Comment (tarampampam) [reply tests/fields/fields_list.test.ts:118]: Seems like not :(

**Comment by tarampampam** [comment]
Anyway - today is deadline for this PR merging. What do you think?

File: tests/fields/fields_list.test.ts (hunk 7 above)
Comment (Reallife) [reply tests/fields/fields_list.test.ts:118]: Made some changes:

```js
case 'float':
    expect(report_field_value).toBeNumber();
    break;
case 'integer':
    expect(Number.isInteger(report_field_value)).toBeTrue();
    break;
```
Now we can be sure that values like `1.3` doesn't pass test for `integer` type at least:

```js
Number.isInteger(1.3) // false
// but
Number.isInteger(1.0) // true =(
```

**Comment by Reallife** [comment]
> Anyway - today is deadline for this PR merging. What do you think?

We can change field's data types from `float` to number in `fields_list.json`. But would be better to do it in next major release(when we plan to remove this file =)). So, i think, we need to release it and we should be more carefully about that fields until `v4.0.0` will be released

File: fields/default/fields_list.json (hunk 6 above)
Comment (tarampampam) [reply fields/default/fields_list.json:3177]: Do not forget to update your changes in changelog file

**Comment by tarampampam** [comment]
Agree

**Pull request merged**